crates/mun_abi/src/type_info.rs

```diff
@@ -229,40 +320,36 @@ macro_rules! impl_primitive_type_info {
                         let guid = Guid::from(type_info_name.as_bytes());
 
                         TypeInfo {
-                            id: TypeId { guid },
                             name: type_info_name.as_ptr(),
                             size_in_bits: (std::mem::size_of::<$ty>() * 8)
                                 .try_into()
                                 .expect("size of T is larger than the maximum allowed ABI size. Please file a bug."),
                             alignment: (std::mem::align_of::<$ty>())
                                 .try_into()
                                 .expect("alignment of T is larger than the maximum allowed ABI size. Please file a bug."),
-                            data: TypeInfoData::Primitive,
+                            data: TypeInfoData::Primitive(guid),
                         }
                     })
                 }
             }
-        )+
-    }
-}
 
-macro_rules! impl_has_type_info_name {
-    ($(
-        $ty:ty => $name:tt
-    ),+) => {
-        $(
-            impl crate::type_info::HasStaticTypeInfoName for $ty {
-                fn type_name() -> &'static std::ffi::CStr {
-                    static TYPE_INFO_NAME: once_cell::sync::OnceCell<std::ffi::CString> = once_cell::sync::OnceCell::new();
-                    let type_info_name: &'static std::ffi::CString = TYPE_INFO_NAME
-                        .get_or_init(|| std::ffi::CString::new($name).unwrap());
-                    type_info_name.as_ref()
+            impl HasStaticTypeId for $ty {
```

Comment:
It might be good to create a `type_id.rs` file that contains the struct definition and static type ID-related code